app/controllers/bulk_taggings_controller.rb

```diff
@@ -0,0 +1,47 @@
+class BulkTaggingsController < ApplicationController
+  def index
+    @bulk_taggings = BulkTagging.all.newest_first
+  end
+
+  def new
+    @bulk_tagging = BulkTagging.new
+  end
+
+  def create
+    @bulk_tagging = BulkTagging.new(bulk_tagging_params)
+    if @bulk_tagging.valid?
+      @bulk_tagging.save!
+      redirect_to bulk_taggings_path
+    else
+      render :new
+    end
+  end
+
+  def show
+    @bulk_tagging = BulkTagging.find(params[:id])
+
+    if @bulk_tagging.tag_mappings.count.zero?
+      @fetch_errors = FetchTaggingSheet.new(@bulk_tagging).run
+    end
+
+    @tag_mappings = @bulk_tagging.tag_mappings.by_content_base_path.by_link_title
+  end
+
+  def refetch
+    @bulk_tagging = BulkTagging.find(params[:bulk_tagging_id])
+    @bulk_tagging.tag_mappings.destroy_all
```

Comment:
If we set the foreign key constraint to cascade delete, we can make this `delete_all`. May prevent us from having to do deletion inside a sidekiq worker.